homeassistant/components/sensor/plex.py

```diff
@@ -63,19 +68,30 @@ def __init__(self, name, plex_url, plex_user, plex_password,
         """Initialize the sensor."""
         from plexapi.myplex import MyPlexAccount
         from plexapi.server import PlexServer
+        import plexapi.exceptions
+
+        cert_session = None
 
         self._name = name
         self._state = 0
         self._now_playing = []
 
-        if plex_token:
-            self._server = PlexServer(plex_url, plex_token)
-        elif plex_user and plex_password:
-            user = MyPlexAccount(plex_user, plex_password)
-            server = plex_server if plex_server else user.resources()[0].name
-            self._server = user.resource(server).connect()
-        else:
-            self._server = PlexServer(plex_url)
+        try:
+            if plex_token:
+                self._server = PlexServer(plex_url, plex_token, cert_session)
+            elif plex_user and plex_password:
+                user = MyPlexAccount(plex_user, plex_password)
+                server = plex_server if plex_server \
+                    else user.resources()[0].name
+                self._server = user.resource(server).connect()
+            else:
+                self._server = PlexServer(plex_url)
+
+            _LOGGER.info("Plex Sensor Configuration done")
+        except (plexapi.exceptions.BadRequest, plexapi.exceptions.Unauthorized,
```

Comment:
Move this try/except to surround the `add_devices` line in setup platform. Ignoring the exception here means we're creating the entity without a valid `self._server`, which the rest of the code doesn't handle.

Reply:
done